nilearn/_utils/niimg.py

```diff
@@ -0,0 +1,157 @@
+"""
+Neuroimaging file input and output.
+"""
+# Author: Gael Varoquaux, Alexandre Abraham, Philippe Gervais
+# License: simplified BSD
+
+import copy
+import gc
+import collections
+from six import string_types
+
+import numpy as np
+import nibabel
+
+from .numpy_conversions import as_ndarray
+
+
+def _safe_get_data(img):
+    """ Get the data in the image without having a side effect on the
```

Comment:
OK. I have 2 versions of nibabel: 1.3.0 (has no cache) and 2.0 (has cache and dataobj) so I don't know which version is targeted by this code. I try to do the mean on 10 images from ADHD. I limited the memory. Here is the script:
```
from nilearn import datasets
from nilearn.image import mean_img
import resource


def limit_memory(maxsize):
    soft, hard = resource.getrlimit(resource.RLIMIT_AS)
    resource.setrlimit(resource.RLIMIT_AS, (maxsize, hard))


limit_memory(165152864 * 5)
data = datasets.fetch_adhd(n_subjects=10)
mean_img(data.func)
```

Given that paths are passed to `mean_img`, the caching system should not change anything between the versions. Both should be able to do the mean given the memory limit as they only need to load images one by one.

The result is that they both endup in memory error after 8 images. There must be a leak somewhere but I can't find out where. I'll keep the code as-is but we should maybe investigate that and add some tests.